lib/mix/tasks/build_task.ex

```diff
@@ -0,0 +1,14 @@
+defmodule Mix.Tasks.Build do
+  use Mix.Task
+
+  @doc """
+  
+  """
+  @shortdoc "Generate Elixir Module from MAVLink dialect XML"
+  @spec run([String.t]) :: :ok
+  @impl Mix.Task
+  
+  def run(_) do
+    MAVLink.Writer.create("test/input/ardupilotmega.xml", "lib/mavlink/dialect/apm.ex", "MAVLink.Dialect.APM")
```

Comment:
I don't know about Tasks.Build, will have to read up on it.

Also more generally I'm not sure I can or should distribute the xml files with this library. Will have a look at pymavgen and what Tridge does. I don't want to create another source of truth for the mavlink definitions, they update every few days, and a mismatch of dialects is a safety issue I'd like users to take responsibility for. Perhaps mavlink message_definitions/v1.0 can be a git submodule or something?

Reply:
I'm not sure how git submodules work when you add a package as a dependency but will look into it. I have however seen https://github.com/bitwalker/timex pull down a bunch of stuff when you add it as a dependency. We could do something similar and put the definitions somewhere, s3 bucket?

Reply:
Couldn't we pull them direct from mavlink github https://github.com/mavlink/mavlink/tree/master/message_definitions/v1.0

Reply:
Although people can create their own definitions, wouldn't want to overwrite them. Someone might just want a robust comms library and not care about any of the drone stuff. Maybe overthinking...

Reply:
Yes we could pull directly, perhaps using tags.

I see so we would want something more dynamic, I will have a look.

Reply:
Yes, more protobufs (or at least compiled from shared definition comms library, if that's what protobufs is) that happens to be used almost exclusively by remote vehicle community (at the moment)